ibis/bigquery/tests/test_compiler.py

```diff
@@ -380,3 +380,57 @@ def test_projection_fusion_only_peeks_at_immediate_parent():
 FROM t3
   CROSS JOIN t3 t4"""
     assert result == expected
+
+
+def test_bool_reducers(alltypes):
+    b = alltypes.bool_col
+    expr = b.mean()
+    result = expr.compile()
+    expected = """\
+SELECT avg(CAST(`bool_col` AS INT64)) AS `mean`
+FROM `ibis-gbq.testing.functional_alltypes`"""
+    assert result == expected
+
+    expr2 = b.sum()
+    result = expr2.compile()
+    expected = """\
+SELECT sum(CAST(`bool_col` AS INT64)) AS `sum`
+FROM `ibis-gbq.testing.functional_alltypes`"""
+    assert result == expected
+
+
+def test_bool_reducers_where(alltypes):
+    b = alltypes.bool_col
+    m = alltypes.month
+    expr = b.mean(where=m > 6)
+    result = expr.compile()
+    expected = """\
+SELECT avg(CASE WHEN `month` > 6 THEN CAST(`bool_col` AS INT64) ELSE NULL END) AS `mean`
+FROM `ibis-gbq.testing.functional_alltypes`"""  # noqa: E501
+    assert result == expected
+
+    expr2 = b.sum(where=((m > 6) & (m < 10)))
```

Comment:
I wanted to confirm is multiple boolean statements would be stored in a single `op().args` slot and it is! Looks like `op().args[1]` is sufficient to unpack the whole where clause.